### PR TITLE
Upstream fstar core changes from cryspen-sandwich.

### LINKED ISSUE #979: Upstream fstar core additions from `cryspen-sandwich` to `main`
Some additions to fstar core were made in the `cryspen-sandwich` branch. It would be great to upstream them to the `main` branch. However some more changes were made in this branch that shouldn't be merged to main.

### PR DESCRIPTION
**Pull request by maximebuyse**
closes #979

### CODE REVIEW

File: proof-libs/fstar/core/Std.Io.Impls.fsti [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module Std.Io.Impls
+val placeholder: unit
```
Comment (W95Psp) [inline proof-libs/fstar/core/Std.Io.Impls.fsti:2]: Why is this needed?

**Review by W95Psp — APPROVED** [review]
Looks good, I only have one interrogation

File: proof-libs/fstar/core/Std.Io.Impls.fsti (hunk 1 above)
Comment (maximebuyse) [reply proof-libs/fstar/core/Std.Io.Impls.fsti:2]: I don't know, we can skip it and add it later if needed

Comment (W95Psp) [reply proof-libs/fstar/core/Std.Io.Impls.fsti]: Can you try to git blame maybe? so that we can maybe correlate the date and author of this change with something in sandwich for instance?

**Comment by maximebuyse** [comment]
I don't understand the CI failure. The core changes are unrelated to the error, and chacha20 type-checking works locally for me (maybe updating the fstar version used by the CI could help?)

**Comment by maximebuyse** [comment]
> I don't understand the CI failure. The core changes are unrelated to the error, and chacha20 type-checking works locally for me (maybe updating the fstar version used by the CI could help?)

After rebasing the CI passes. Let's merge.
